Добавлены ViewModel для списка фильмов и детельной информации.

demqn83 opened the pull request.

demqn83 committed:
Добавлены ViewModel для списка фильмов и детельной информации.

demqn83 committed:
ViewModelProvider для опияания фильма.

demqn83 committed:
Получение контекста в классах Factory.

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -9,24 +9,43 @@ import android.widget.ImageView
 import android.widget.RatingBar
```
![image](https://user-images.githubusercontent.com/11761610/103235327-ee5ab500-4952-11eb-9f0e-01a5147aa4bb.png)
Круто будет если разложишь классы
по пакетам)

ivanlardis commented on app/src/main/java/ru/demqn/appname/MovieDetailsViewModelFactory.kt:
```diff
@@ -0,0 +1,16 @@
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel?> create(modelClass: Class<T>): T = when (modelClass) {
+        MovieDetailsViewModel::class.java -> MovieDetailsViewModel(MovieUtil(context), movieId)
```
Вот тут все ок сейчас
мы потом узнаем о других классах архитектуры и добавятся они
а сейчас пока все ок)

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -9,24 +9,43 @@ import android.widget.ImageView
+        savedInstanceState: Bundle?
+    ): View? {
 
         val movieIdKey = requireNotNull(arguments?.getInt(MOVIE_ID_KEY))
```
тут movieId а не key

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -9,24 +9,43 @@ import android.widget.ImageView
 
         val movieIdKey = requireNotNull(arguments?.getInt(MOVIE_ID_KEY))
         val view = inflater.inflate(R.layout.fragment_movies_details, container, false)
+//        movieDetailsViewModel = MovieDetailsViewModel(movieIdKey)
```
удаляем такие комменты
код же грязнится

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -9,24 +9,43 @@ import android.widget.ImageView
+        movieDetailsViewModel = ViewModelProvider(this, MovieDetailsViewModelFactory(requireContext(), movieIdKey)).get(MovieDetailsViewModel::class.java)
+
+        movieDetailsViewModel.movie.observe(this.viewLifecycleOwner, this::updMovie)
+        movieDetailsViewModel.getMovie()
```
конкретно тут movieId
лучше передавать в getMovie(movieId)

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -9,24 +9,43 @@ import android.widget.ImageView
-import ru.demqn.appname.data.MovieUtil
 
 
 class FragmentMoviesDetails : Fragment() {
```
![image](https://user-images.githubusercontent.com/11761610/103236458-02ec7c80-4956-11eb-9088-88e08a3856ae.png)
Переписал по тем моментам которые уже обсудили сегодня в зоом

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesList.kt:
```diff
@@ -6,26 +6,23 @@ import android.view.LayoutInflater
-import ru.demqn.appname.data.MovieUtil
 
 
 class FragmentMoviesList : Fragment() {
```
давай перепишем по анологии сверзу

ivanlardis commented on app/src/main/java/ru/demqn/appname/MovieDetailsViewModel.kt:
```diff
@@ -0,0 +1,22 @@
+
+    fun getMovie() {
+        viewModelScope.launch {
+//            delay(5000)
```
давай удалять такое прям режет глаз

ivanlardis commented on app/src/main/java/ru/demqn/appname/MovieDetailsViewModel.kt:
```diff
@@ -0,0 +1,22 @@
+    private val _mutableMovie = MutableLiveData<Movie>()
+    val movie: LiveData<Movie> get() = _mutableMovie
+
+    fun getMovie() {
```
договорились movieId передавать тут

demqn83 committed:
Удаление лишних комментариев.
getMovie с параметром movieId.
Обновление полей вью, через поля фрагмента.
Вынос строк кода в отдельные функции.

demqn83 committed:
Обработка полученного списка фильмов, для передачи готового списка для отображения без необходимости дополнительной обработки.
